src/main/kotlin/com/exactpro/th2/check1/metrics/BufferMetric.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.check1.metrics
+
+import com.exactpro.th2.check1.SessionKey
+import com.exactpro.th2.check1.configuration.Check1Configuration
+import com.exactpro.th2.check1.utils.dec
+import com.exactpro.th2.check1.utils.inc
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.message.toJson
+import com.exactpro.th2.common.metrics.DEFAULT_DIRECTION_LABEL_NAME
+import com.exactpro.th2.common.metrics.DEFAULT_SESSION_ALIAS_LABEL_NAME
+import io.prometheus.client.Gauge
+import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+import mu.KotlinLogging
+import org.openjdk.jol.info.GraphLayout
+import java.util.Queue
+import java.util.LinkedList
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.TimeUnit
+
+object BufferMetric : AutoCloseable {
+    private val logger = KotlinLogging.logger {}
+
+    private val averageReceivedMessagesSizeMetric: Gauge = Gauge
```

Comment:
I don't think we need this metric. We can use the `increase` function in the Grafana to calculate the average size using the `th2_check1_actual_cache_size` and `th2_check1_actual_cache_number` metrics